### PR TITLE
Ensure header is defined in INFO when checking for Number=A.

### LINKED ISSUE #318: Key error when accessing undefined header
The new logic for getting `Number=A` fields from the header should check whether the key exists in the defined header: https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/libs/processed_variant.py#L230

Note that the pipeline would fail in any case as the header is not defined (b/c the field is missing in the BigQuery schema), but the error message would be [more descriptive](https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/libs/bigquery_row_generator.py#L190).

### PR DESCRIPTION
**Pull request by arostamianfar**
Tested: manual run using an undefined header. Note that this doesn't get caught in integration testing as the pipeline is still supposed to fail, just with a better exception ;)

Fixes #318

### CODE REVIEW

File: gcp_variant_transforms/libs/processed_variant.py [hunk 1]
```diff
@@ -247,8 +247,9 @@ def create_processed_variant(self, variant):
         self._annotation_processor.add_annotation_data(
             proc_var, key, variant_info_data)
       elif (self._split_alternate_allele_info_fields and
-            self._header_fields.infos[key][_HeaderKeyConstants.NUM] ==
-            vcf.parser.field_counts[_FIELD_COUNT_ALTERNATE_ALLELE]):
+            key in self._header_fields.infos and
```
[the remark this answers is not in the document]
Comment (arostamianfar) [reply gcp_variant_transforms/libs/processed_variant.py]: I think the combination of `.get` and None is more convoluted and it does another implicit boolean check for None. In general, `.get` is only useful if you want to return another value if the key does not exist, but checking for membership should be done with `in`. I basically agree with [this answer](https://stackoverflow.com/a/46509366).

[the remark this answers is not in the document]
Comment (samanvp) [reply gcp_variant_transforms/libs/processed_variant.py]: Oh I see! I honestly did not know about this application of `in`. Definitely this looks better than `get()`.
I have a bunch of `get()` in my code that I exactly do the same thing as you are doing here. I need to remember to update them :D

**Comment by arostamianfar** [comment]
@samanvp : actually, your comment made me realize that we have duplicated logic for this method :) I created a helper function for this and removed the other one. PTAL. Thanks!

File: gcp_variant_transforms/libs/processed_variant.py [hunk 2]
```diff
@@ -358,13 +355,19 @@ def info_is_in_alt_bases(self, info_field_name):
...
     is_annotation = info_field_name in self._annotation_field_set
     return is_per_alt_info or is_annotation
 
+  def _is_per_alt_info_field(self, info_field_name):
+    # type: (str) -> bool
+    """Returns true iff `info_field_name` is defined as having Number=A."""
+    return (
+        self._split_alternate_allele_info_fields and
+        info_field_name in self._header_fields.infos and
+        self._header_fields.infos[info_field_name][_HeaderKeyConstants.NUM] == (
```
[the remark this answers is not in the document]
Comment (pgrosu) [reply gcp_variant_transforms/libs/processed_variant.py]: @samanvp This is a specific predicate function of interpretation - which is always good to use - that in this case it is equal to the number of alternate alleles in a given record, which is always of that type.  This link below might be helpful:

https://github.com/jamescasbon/PyVCF/blob/master/vcf/parser.py#L63-L69

[the remark this answers is not in the document]
Comment (arostamianfar) [reply gcp_variant_transforms/libs/processed_variant.py]: @samanvp : we can be sure that that one exists :) It's a predefined `namedtuple`

**Pull request merged**